### PR TITLE
simplify testing of ServiceHelper

### PR DESCRIPTION
**Pull request by russgold**
This is a step towards addressing OWLS-72379 and OWLS-72614.

The previous tests not only used the old mock-style (specifying the exact k8s calls to be made), they looked for services to be replaced after the life ones were modified. Given our experience with OpenShift, that it not a viable approach, so the new tests accomplish much the same by changing the domain to generate revised services. I have also simplified the testing and handled common cases via parameterization.

In doing this work, it became clear that there are some problems with the implementation of the External Service:

1. The "labels" and "annotations" fields of adminService were never implemented
2. The service is being added to the domain presence as a server service for the admin server

In addition, it is not clear to me that the correct node ports are being created, and I would like an explanation of the intent. I have marked a number of tests disabled for the external service, pending some clarification and corrections.

### CODE REVIEW

File: docs/domains/Domain.json [hunk 1]
```diff
@@ -51,6 +51,14 @@
           "items": {
             "$ref": "#/definitions/Channel"
           }
+        },
```
Comment (rjeberhard) [inline docs/domains/Domain.json]: Just a heads-up that we can't merge this until we merge the code to upgrade the Domain CRD version.  Even though you would just be adding optional fields (which is good), the issue is that unless we version the CRD then a 2.0.1 operator would revert the change to the CRD by writing its version.

File: model/src/main/java/oracle/kubernetes/weblogic/domain/ClusterConfigurator.java [hunk 2]
```diff
@@ -1,4 +1,4 @@
-// Copyright 2018, Oracle Corporation and/or its affiliates.  All rights reserved.
+// Copyright 2018,2019, Oracle Corporation and/or its affiliates.  All rights reserved.
```
Comment (rjeberhard) [inline model/src/main/java/oracle/kubernetes/weblogic/domain/ClusterConfigurator.java]: Mark has been teaching me to be more pedantic about these copyrights :(...  Apparently, there needs to be a space as in, "Copyright 2018, 2019,"

File: model/src/main/java/oracle/kubernetes/weblogic/domain/ServiceConfigurator.java [hunk 3]
```diff
@@ -0,0 +1,12 @@
+// Copyright 2019 Oracle Corporation and/or its affiliates.  All rights reserved.
```
Comment (rjeberhard) [inline model/src/main/java/oracle/kubernetes/weblogic/domain/ServiceConfigurator.java]: "Copyright 2019,"

File: model/src/main/java/oracle/kubernetes/weblogic/domain/v2/AdminServerSpec.java [hunk 4]
```diff
@@ -0,0 +1,17 @@
+// Copyright 2019 Oracle Corporation and/or its affiliates.  All rights reserved.
```
Comment (rjeberhard) [inline model/src/main/java/oracle/kubernetes/weblogic/domain/v2/AdminServerSpec.java]: "2019,"

File: model/src/main/java/oracle/kubernetes/weblogic/domain/v2/AdminServerSpecV2Impl.java [hunk 5]
```diff
@@ -1,11 +1,19 @@
-// Copyright 2018, Oracle Corporation and/or its affiliates.  All rights reserved.
+// Copyright 2018,2019, Oracle Corporation and/or its affiliates.  All rights reserved.
```
Comment (rjeberhard) [inline model/src/main/java/oracle/kubernetes/weblogic/domain/v2/AdminServerSpecV2Impl.java]: "2018, 2019,"

File: model/src/main/java/oracle/kubernetes/weblogic/domain/v2/ServerSpec.java [hunk 6]
```diff
@@ -1,254 +1,132 @@
-// Copyright 2018, 2019, Oracle Corporation and/or its affiliates.  All rights reserved.
+// Copyright 2019 Oracle Corporation and/or its affiliates.  All rights reserved.
```
Comment (rjeberhard) [inline model/src/main/java/oracle/kubernetes/weblogic/domain/v2/ServerSpec.java]: "2019,"

File: model/src/test/java/oracle/kubernetes/weblogic/domain/ChannelMatcher.java [hunk 7]
```diff
@@ -0,0 +1,47 @@
+// Copyright 2019 Oracle Corporation and/or its affiliates.  All rights reserved.
```
Comment (rjeberhard) [inline model/src/test/java/oracle/kubernetes/weblogic/domain/ChannelMatcher.java]: "2019,"

File: operator/src/main/java/oracle/kubernetes/operator/wlsconfig/WlsDynamicServerConfig.java [hunk 8]
```diff
@@ -1,4 +1,4 @@
-// Copyright 2018, Oracle Corporation and/or its affiliates.  All rights reserved.
+// Copyright 2018,2019, Oracle Corporation and/or its affiliates.  All rights reserved.
```
Comment (rjeberhard) [inline operator/src/main/java/oracle/kubernetes/operator/wlsconfig/WlsDynamicServerConfig.java]: "2018, 2019,"

File: operator/src/test/java/oracle/kubernetes/operator/helpers/ServiceHelperDeletionTest.java [hunk 9]
```diff
@@ -0,0 +1,91 @@
+// Copyright 2019 Oracle Corporation and/or its affiliates.  All rights reserved.
```
Comment (rjeberhard) [inline operator/src/test/java/oracle/kubernetes/operator/helpers/ServiceHelperDeletionTest.java]: "2019, "

File: operator/src/test/java/oracle/kubernetes/operator/helpers/ServiceHelperTest.java [hunk 10]
```diff
@@ -1,86 +1,85 @@
-// Copyright 2018, 2019, Oracle Corporation and/or its affiliates.  All rights reserved.
+// Copyright 2019 Oracle Corporation and/or its affiliates.  All rights reserved.
```
Comment (rjeberhard) [inline operator/src/test/java/oracle/kubernetes/operator/helpers/ServiceHelperTest.java]: "2019,"

File: operator/src/test/java/oracle/kubernetes/operator/helpers/ServiceHelperTestBase.java [hunk 11]
```diff
@@ -0,0 +1,42 @@
+// Copyright 2019 Oracle Corporation and/or its affiliates.  All rights reserved.
```
Comment (rjeberhard) [inline operator/src/test/java/oracle/kubernetes/operator/helpers/ServiceHelperTestBase.java]: "2019,"

File: operator/src/test/java/oracle/kubernetes/operator/wlsconfig/WlsDynamicServerConfigTest.java [hunk 12]
```diff
@@ -1,4 +1,4 @@
-// Copyright 2018, Oracle Corporation and/or its affiliates.  All rights reserved.
+// Copyright 2018,2019, Oracle Corporation and/or its affiliates.  All rights reserved.
```
Comment (rjeberhard) [inline operator/src/test/java/oracle/kubernetes/operator/wlsconfig/WlsDynamicServerConfigTest.java]: "2018, 2019,"

**Comment by rjeberhard** [comment]
Wow!  This is a big, very good change.  Let's fix the issues with copyright statements and then put this in right after the CRD version update.

**Comment by russgold** [comment]
Replaced by PR 933
